lib/src/kbasesearchengine/search/ObjectData.java

```diff
@@ -33,7 +35,9 @@
     private final Optional<Object> parentData;
     private final Optional<Object> data;
     private final Map<String, String> keyProps;
-    
+    private final Map<String, ArrayList> highlight;
```

Comment:
No non-parameterized Generics - this applies in several places in this class.

Reply:
Do you mean I shouldn't do something like 
private Map<String, ArrayList> highlight = new HashMap<>();

Reply:
No, it means you sholdn't use generic classes without parameterizing them.

Reply:
so I should do something like this? 
private Map<String, ArrayList<String>> highlight = new HashMap<>();

Reply:
No, ArrayList is a generic class, and it's not parameterized.

Reply:
wtf wrong comment posted. I meant

private Map<String, ArrayList<String>> highlight = new HashMap<>();

Reply:
interesting so in github you need to escape <> characters otherwise the inside doesnt show

Reply:
yes. also as I mentioned elsewhere don't specify the list implementation

Reply:
what do you mean by not specifying list implementation? Do List<String> instead of ArrayList<String>?

Reply:
Right, unless you need a property of a specific implementation, which you usually don't

Reply:
okay it makes sense now